cmd/login/runner_test.go

```diff
@@ -0,0 +1,111 @@
+package login
+
+import (
+	"bytes"
+	"strconv"
+	"testing"
+
+	"github.com/giantswarm/microerror"
+	"github.com/spf13/cobra"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+)
+
+func TestLogin(t *testing.T) {
+	testCases := []struct {
+		name        string
+		startConfig *clientcmdapi.Config
+		mcArg       string
+		expectError *microerror.Error
+	}{
+		{
+			name:        "case 0",
+			startConfig: &clientcmdapi.Config{},
+			mcArg:       "codename",
+			expectError: contextDoesNotExistError,
+		},
+		{
+			name:        "case 1",
+			startConfig: &clientcmdapi.Config{},
+			mcArg:       "gs-codename",
+			expectError: contextDoesNotExistError,
+		},
+		{
+			name:        "case 2",
+			startConfig: createValidTestConfig(),
+			mcArg:       "gs-codename",
+		},
+		{
+			name:        "case 3",
+			startConfig: createValidTestConfig(),
+			mcArg:       "codename",
+		},
+		{
+			name:        "case 4",
+			startConfig: createValidTestConfig(),
+			mcArg:       "othercodename",
+			expectError: contextDoesNotExistError,
+		},
+		{
+			name:        "case 5",
+			startConfig: createValidTestConfig(),
+			mcArg:       "gs-othercodename",
+			expectError: contextDoesNotExistError,
+		},
+		{
+			name:        "case 6",
+			startConfig: createValidTestConfig(),
+			mcArg:       "https://anything.com",
+			expectError: unknownUrlError,
+		},
+	}
+
+	for i, tc := range testCases {
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
+			r := runner{
+				k8sConfigAccess: &clientcmd.ClientConfigLoadingRules{
+					ExplicitPath: "test/config.yaml",
```

Comment:
By convention, folders for this sort of fixtures should be named `testdata`. This way they are not interpreted as package names.

See https://pkg.go.dev/cmd/go#hdr-Test_packages